src/components/settings/SettingsForm.tsx

```diff
@@ -0,0 +1,316 @@
+import classNames from 'classnames';
+import { CopyIcon, EyeIcon, EyeOffIcon } from 'lucide-react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useNotification } from '../../hooks/useNotification';
+import { useSessionStorage } from '../../hooks/useSessionStorage';
+import { decode } from '../../node/encoding';
+import { setSettingsApiKey } from '../../node/settings';
+import { getSettings } from '../../node/settings';
+import type { Pin } from '../../types';
+import { handleLinkClick } from '../../utils';
+import Button from '../common/Button';
+import Notification from '../common/Notification';
+import Description from '../typography/Description';
+import Label from '../typography/Label';
+import PageHeading from '../typography/PageHeading';
+
+export default function SettingsForm() {
+  const [showApiKey, setShowApiKey] = useState(false);
+  const [showPin, setShowPin] = useState(false);
+
+  const [loading, setLoading] = useState(false);
+  const { notification, notifySuccess, notifyError, clearNotification } =
+    useNotification();
+  const { setItem, getItem } = useSessionStorage();
+
+  const [apiKey, setApiKey] = useState<string>();
+  const [pin, setPin] = useState<Pin>();
+  const [confirmPin, setConfirmPin] = useState<Pin>();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    setLoading(true);
+
+    if (apiKey) {
+      if (pin && Object.values(pin).some((value) => !value)) {
+        notifyError('All digits must be filled');
+        setLoading(false);
+        return;
+      }
+
+      if (pin && JSON.stringify(pin) !== JSON.stringify(confirmPin)) {
+        notifyError('Pins do not match');
+        setLoading(false);
+        return;
+      }
+
+      try {
+        await setSettingsApiKey(apiKey, pin);
+        notifySuccess('Settings saved successfully');
+        if (pin) {
+          setItem('pin', JSON.stringify(pin));
+          setPin(undefined);
+          setConfirmPin(undefined);
+        }
+        setLoading(false);
+      } catch (error) {
+        notifyError('Failed to save settings', (error as Error).message);
+        setLoading(false);
+        setPin(undefined);
+        setConfirmPin(undefined);
+      }
+    }
+  };
+
+  useEffect(() => {
+    const getApiKey = async () => {
+      const settings = await getSettings();
+      let apiKey = settings.apiKey;
+
+      if (apiKey && settings.hasPin && getItem('pin')) {
+        const pin = getItem('pin') as string;
+        const parsedPin = JSON.parse(pin);
+        const secret = `${parsedPin.first}${parsedPin.second}${parsedPin.third}${parsedPin.fourth}`;
+
+        apiKey = decode(secret, apiKey);
```

Comment:
to display to a user if he has it already setup, to be able to see it, if he wants. But if we are going with that approach above, to not show it at all when it is already setup, then this will be removed

this comment: https://github.com/plainly-videos/after-effects-plugin/pull/16#discussion_r1895561336

Reply:
Yup